mongo/job_store.go

```diff
@@ -92,6 +96,45 @@ func (m *JobStore) CreateJob(ctx context.Context, id string) (job models.Job, er
 	return newJob, nil
 }
 
+// CreateTask creates a new task, for the given API and job ID, in the collection, and assigns default values to its attributes
+func (m *JobStore) CreateTask(ctx context.Context, jobID string, nameOfApi string, numDocuments int) (task models.Task, err error) {
+	log.Info(ctx, "creating task in mongo DB", log.Data{"jobID": jobID, "nameOfApi": nameOfApi, "numDocuments": numDocuments})
+
+	// If an empty job id was passed in, return an error with a message.
+	if jobID == "" {
+		return models.Task{}, ErrEmptyIDProvided
+	}
+
+	s := m.Session.Copy()
+	defer s.Close()
+
+	// Check that the jobs collection contains the job that the task will be part of
+	var jobToFind models.Job
+	jobToFind.ID = jobID
+	err = s.DB(m.Database).C(m.JobsCollection).Find(bson.M{"_id": jobID}).One(&jobToFind)
+	if err != nil {
+		log.Error(ctx, "error finding job for task", err)
+		if err == mgo.ErrNotFound {
+			return models.Task{}, ErrJobNotFound
+		} else {
+			return models.Task{}, err
```

Comment:
Would be nice to have context describing what function call failed here e.g.:
return models.Task{}, fmt.Errorf("failed to find job: %w", err)

Reply:
The ErrJobNotFound describes the failed to find job - the other error is there if something unexpected happens. I've added something to say that now.